### PR TITLE
[Develop] Prevent updates to clusters that include changes to /home

### PR DESCRIPTION
**Pull request by dreambeyondorange**
Currently, PCluster does not support updates to a cluster's home directory.

### Tests
* Tested updates to clusters that have existing /home mounts in `SharedStorage` and those that do not have /home mounts

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by himani2411 — DISMISSED** [review]

File: cli/src/pcluster/config/update_policy.py [hunk 1]
```diff
@@ -227,7 +227,18 @@ def is_stop_required_for_shared_storage(change):
     return False
 
 
+def is_home_change(change):
+    if change.key == "SharedStorage":
+        if (change.old_value and change.old_value.get("MountDir") in ["home", "/home"]) or (
```
Comment (gmarciani) [inline cli/src/pcluster/config/update_policy.py:232]: Minor: if you see that we are duplicating the list [home, /home] also elsewhere I suggest to define a constant to reduce duplication and reduce risks of errors (specifying only one of the two in other parts of the code)

Comment (gmarciani) [inline cli/src/pcluster/config/update_policy.py:232]: With this condition we will prevent to change from home to /home and viceversa, even if /home and home are semantically equivalent for us. Putting this comment just to verify that this is expected.

File: cli/src/pcluster/config/update_policy.py [hunk 2]
```diff
@@ -318,6 +329,8 @@ def condition_checker_managed_fsx(change, patch):
 
 
 def actions_needed_shared_storage_update(change, patch):
+    if is_home_change(change):
+        return "Please revert any changes to the /home mount"
```
Comment (gmarciani) [inline cli/src/pcluster/config/update_policy.py:333]: What about capture the scenario with unit tests?

File: cli/src/pcluster/config/update_policy.py (hunk 1 above)
Comment (dreambeyondorange) [reply cli/src/pcluster/config/update_policy.py:232]: Actually, I think changing from /home to home doesn't result in a change to evaluate

Comment (dreambeyondorange) [reply cli/src/pcluster/config/update_policy.py:232]: Currently this is not used anywhere else in the codebase

File: cli/src/pcluster/config/update_policy.py (hunk 2 above)
Comment (dreambeyondorange) [reply cli/src/pcluster/config/update_policy.py:333]: Good point, I will add these!

**Comment by dreambeyondorange** [comment]
Added unit tests!

**Review by gmarciani — APPROVED** [review]

**Pull request merged**